python/pyspark/pandas/tests/test_dataframe.py

```diff
@@ -5614,6 +5614,40 @@ def test_at_time(self):
         with self.assertRaisesRegex(TypeError, "Index must be DatetimeIndex"):
             psdf.at_time("0:15")
 
+    def test_combine_first(self):
```

Comment:
Could you also try with the DataFrames in the same anchor?

E.g.,:

```py
pdf = pd.DataFrame({("X", "A"): [None, 0], ("X", "B"): [4, None], ("Y", "C"): [3, 3], ("Y", "B"): [1, 1]})
pdf1 = pdf["X"]
pdf2 = pdf["Y"]
psdf = ps.from_pandas(pdf)
psdf1 = psdf["X"]
psdf2 = psdf["Y"]

...
```

Hopefully the result can preserve the same anchor. I'm not sure whether we can or not, though.

Reply:
Btw, the following tests should be in `test_ops_on_diff_frames.py`?

Reply:
Good point, the test is moved to `test_ops_on_diff_frames.py`.

I may need some more time to make frames with the same anchor work.

Reply:
Frames with the same anchor work with the current approach, but the anchor is lost afterward. I guess that's not acceptable. I will see if there is an optimized approach.

Reply:
`Series.combine_first` might be a good reference?

Reply:
Let me take a look then, thanks!